tests/e2e/utils/functions.py

```diff
@@ -1,14 +1,27 @@
-from concurrent.futures import ThreadPoolExecutor
+import time
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import List
 
 from tests.e2e.utils import data_fetcher  # noqa F401
 
 
 def execute_lambdas_in_parallel(function_name: str, lambdas_arn: list, arguments: str):
+    def f(fname: str, farn: str, fargs: str):
+        return eval(fname)(farn, fargs)
+
     result_list = []
     with ThreadPoolExecutor() as executor:
-        running_tasks = executor.map(lambda exec: eval(function_name)(*exec), [(arn, arguments) for arn in lambdas_arn])
+        running_tasks: List[Future] = []
+        for arn in lambdas_arn:
+            # Sleep 0.5, 1, 1.5, ... seconds between each invocation. This way
+            # we can guarantee that lambdas are executed in parallel, but they are
```

Comment:
not sure I follow - how will they be called in parallel if you're adding them to the executor within a 0.5s interval? What was the issue before?

Reply:
Use case: we want to fire two lambdas at the same time for the idempotency test. Each Lambda will block for 3 seconds. We want the first lambda to block for 3 seconds, and the second Lambda to throw with "execution is already running".

Before this change, both Lambdas will be fired at the same time, and due to networking jitter and other factors, we couldn't assume the order that they will actually be run. So in this case we guarantee that, while they will be called in parallel, each successive invocation will be slightly shifted in time to make sure we guarantee the output order. Does this make sense?

Reply:
It does - thank you!!

For the future (not in this PR), I'd suggest renaming or having a parameter for that purpose so we don't accidentally use it for another use case that might incur side effects in assertion (e.g., execution finishing fast enough).